CMakeLists.txt

```diff
@@ -39,6 +39,11 @@
 # (From Daniel Stenberg) The gcc command line use neither -g nor any -O options. As a developer, I also treasure our configure scripts's --enable-debug option that sets a long range of "picky" compiler options.
 cmake_minimum_required(VERSION 3.2...3.16 FATAL_ERROR)
 
+# Revert CMake bug triggered by curl's defined max CMake policy version, see https://gitlab.kitware.com/cmake/cmake/-/issues/21288
```

Comment:
I don't think calling it a CMake bug is right - it's problematic, sure but very much intentional behaviour on the CMake side.
Just use this comment to explain what and why is being set here so next person does not have to scratch their head looking what the CMP0091 is.